docs/developer-guide/writing-actions-reducers.md

```diff
@@ -0,0 +1,203 @@
+
+> ┻┳|  
+> ┳┻| _  
+> ┻┳| •.•) 💬 *"Hey, Checkout this awesome documentation for actions and reducers!"*  
+> ┳┻|⊂ﾉ     
+> ┻┳|  
+
+## What are actions?
+Quoting the redux [documentation](https://redux.js.org/basics/actions) they are:
+> Actions are payloads of information that send data from your application to your store.
+
+They are simply plain JavaScript objects
+```
+/* trigger the panning action of the map to a center point */
+const center = [42.3, 36.5];
+export const PAN_TO = 'MAP:PAN_TO';
+{
+    type: PAN_TO,
+    center
+}
+```
+They must have type property, typically a constant with a string value, but any other properties are optional
+
+## Why we use them
+We need them to trigger changes to the application's store via reducers.
+To do that we use Action Creators
+
+## Action Creators
+They are simply function that returns actions objects
+```
+const defaultValue = [42.3, 36.5];
+/*
+ * by convention, use an initial name (the action filename)
+ * in order to describe better the action type, in this case MAP
+ * separated by a colon : and the action constant name
+*/
+export const PAN_TO = 'MAP:PAN_TO';
+export const panTo = (center = defaultValue) => ({
+    type: PAN_TO,
+    center
+});
+```
+**Note:** Stick to es6 import/export module system and when possible provide a default value for the parameters
+
+These action creators are used in the connected components or in [MapStore2 plugins](../plugins-howto/#connectedsamplejsx-2)
+But actions by themselves are not enough we need Reducers that intercepts those actions and change the state accordingly.
+
+**Note:** Remember to put all the actions .js files in the web/client/actions folder or in js/actions if you are working with custom plugins
+
+## Reducers
+
+Again quoting redux [documentation](https://redux.js.org/basics/reducers) they are:
+> Reducers specify how the application's state changes in response to actions sent to the store.
+
+Reducers are pure function that takes the previous state and an action and returns a new state
+> (previousState, action) => newState
+
+let's see an example:
+```
+ // @mapstore is an alias for dir_name/web/client (see webpack.config.js)
+ import {PAN_TO} from '@mapstore/actions/map';
+
+export function map( (state, action) => {
+        switch(action.type) {
+            case PAN_TO: {
+                return {
+                    ...state,
+                    center
+                };
+            }
+            default return state;
+        }
+    }
+);
+```
+As you can see we are changing the center of the map that triggers the panning action of the mapping library
+
+And that's it we have wrote an action and a reducers that make the map panning around.
+
+**Note:** Remember to put all the reducers .js files in the web/client/reducers folder or in js/reducers if you are working with custom plugins
+
+## Advanced usage and tips
+Sometimes you need to change a value of an item which is store inside an array or in a nested object.
```

Comment:
Sometimes you need to change a value of an item which is stored inside an array or in a nested object.